private/bufpkg/bufplugin/bufplugin.go

```diff
@@ -60,3 +63,122 @@ func NewPlugin(
+		goConfig.RuntimeLibraries = make([]*registryv1alpha1.GoConfig_RuntimeLibrary, 0, len(pluginRuntime.Go.Deps))
+		for _, dependency := range pluginRuntime.Go.Deps {
+			goConfig.RuntimeLibraries = append(goConfig.RuntimeLibraries, GoRuntimeDependencyToProtoGoRuntimeLibrary(dependency))
+		}
+		runtimeConfig.RuntimeConfig = &registryv1alpha1.RuntimeConfig_GoConfig{GoConfig: goConfig}
+	} else if pluginRuntime.NPM != nil {
+		npmConfig := &registryv1alpha1.NPMConfig{}
+		npmConfig.RuntimeLibraries = make([]*registryv1alpha1.NPMConfig_RuntimeLibrary, 0, len(pluginRuntime.NPM.Deps))
+		for _, dependency := range pluginRuntime.NPM.Deps {
+			npmConfig.RuntimeLibraries = append(npmConfig.RuntimeLibraries, NPMRuntimeDependencyToProtoNPMRuntimeLibrary(dependency))
+		}
+		runtimeConfig.RuntimeConfig = &registryv1alpha1.RuntimeConfig_NpmConfig{NpmConfig: npmConfig}
+	}
+	return runtimeConfig
+}
+
+func ProtoRuntimeConfigToPluginRuntime(config *registryv1alpha1.RuntimeConfig) *bufpluginconfig.RuntimeConfig {
+	if config == nil {
+		return nil
+	}
+	runtimeConfig := &bufpluginconfig.RuntimeConfig{}
+	if config.GetGoConfig() != nil {
+		goConfig := &bufpluginconfig.GoRuntimeConfig{}
+		goConfig.MinVersion = config.GetGoConfig().MinimumVersion
+		goConfig.Deps = make([]*bufpluginconfig.GoRuntimeDependencyConfig, 0, len(config.GetGoConfig().RuntimeLibraries))
+		for _, library := range config.GetGoConfig().RuntimeLibraries {
+			goConfig.Deps = append(goConfig.Deps, ProtoGoRuntimeLibraryToGoRuntimeDependency(library))
+		}
+		runtimeConfig.Go = goConfig
+	} else if config.GetNpmConfig() != nil {
+		npmConfig := &bufpluginconfig.NPMRuntimeConfig{}
+		npmConfig.Deps = make([]*bufpluginconfig.NPMRuntimeDependencyConfig, 0, len(config.GetNpmConfig().RuntimeLibraries))
+		for _, library := range config.GetNpmConfig().RuntimeLibraries {
+			npmConfig.Deps = append(npmConfig.Deps, ProtoNPMRuntimeLibraryToNPMRuntimeDependency(library))
+		}
+		runtimeConfig.NPM = npmConfig
+	}
+	return runtimeConfig
+}
+
+func GoRuntimeDependencyToProtoGoRuntimeLibrary(config *bufpluginconfig.GoRuntimeDependencyConfig) *registryv1alpha1.GoConfig_RuntimeLibrary {
+	return &registryv1alpha1.GoConfig_RuntimeLibrary{
+		Module:  config.Module,
+		Version: config.Version,
+	}
+}
+
+func ProtoGoRuntimeLibraryToGoRuntimeDependency(config *registryv1alpha1.GoConfig_RuntimeLibrary) *bufpluginconfig.GoRuntimeDependencyConfig {
+	return &bufpluginconfig.GoRuntimeDependencyConfig{
+		Module:  config.Module,
+		Version: config.Version,
+	}
+}
+
+func NPMRuntimeDependencyToProtoNPMRuntimeLibrary(config *bufpluginconfig.NPMRuntimeDependencyConfig) *registryv1alpha1.NPMConfig_RuntimeLibrary {
+	return &registryv1alpha1.NPMConfig_RuntimeLibrary{
+		Package: config.Package,
+		Version: config.Version,
+	}
+}
+
+func ProtoNPMRuntimeLibraryToNPMRuntimeDependency(config *registryv1alpha1.NPMConfig_RuntimeLibrary) *bufpluginconfig.NPMRuntimeDependencyConfig {
```

Comment:
It looks like we're only using this in this package - let's make sure to un-export all the functions that don't need to be exported. This will apply to a variety of other functions in this file.